docs/go-selectors.md

```diff
@@ -0,0 +1,120 @@
+---
+id: go-selectors
+title: Go Selectors
+sidebar_label: Selectors
+---
+
+## Overview
+
+In Go, the `select` statement lets a goroutine wait on multiple communication operations.
+A `select` **blocks until one of its cases can run**, then it executes that case.
+It chooses one at random if multiple are ready.
+Temporal's Go SDK `Selector`s are similar, but can block on sending and receiving from `Channel`s as well as `Future` deferred work.
```

Comment:
I would rewrite this paragraph to emphasize that Go select cannot be used in Temporal workflows due to determinism. That's why Selector is provided as a replacement. As a bonus, it has additional features like the ability to listen on a Future.